cypress/e2e/bookStore.cy.js

```diff
@@ -1,11 +1,81 @@
 /// <reference types='cypress' />
 
-describe('Book Store app', () => {
-  before(() => {
-    
+const {generateUser} = require('../support/generateUser');
+
+describe('Login', () => {
+  beforeEach(() => {
+    cy.visit('https://demoqa.com/login');
+  });
+
+  it('should contain Login form', () => {
+    cy.findByPlaceholder('UserName').should('exist');
+    cy.findByPlaceholder('Password').should('exist');
+    cy.findById('newUser').should('contain.text', 'New User');
+    cy.findById('login').should('contain.text', 'Login');
+  });
+
+  it('should provide the ability successfuly login', () => {
+    cy.registerUser().then((user) => {
+      cy.findByPlaceholder('UserName').type(user.userName);
+      cy.findByPlaceholder('Password').type(user.password);
+      cy.findById('login').click();
+      cy.findById('userName-value').should('contain.text', user.userName);
+      cy.url('https://demoqa.com/profile')
+    });
+  });
+});
+
+describe('Add book', () => {
+  beforeEach(() => {
+    cy.viewport(550, 750)
```

Comment:
Why do we need to test the app on a small screen?

Reply:
Because when I tried to run my test without the screen's  size limitation, my test fail (Some buttons doesn't visible because the page doesn't scroll)